calligraphy/src/main/java/uk/co/chrisjenx/calligraphy/CalligraphyFactory.java

```diff
@@ -180,12 +180,16 @@ void onViewCreatedInternal(View view, final Context context, AttributeSet attrs)
             parent.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
                 @Override
                 public void onGlobalLayout() {
-                    // No children, do nuffink!
-                    if (parent.getChildCount() <= 0) return;
-                    // Process children, defer draw as it has set the typeface.
-                    for (int i = 0; i < parent.getChildCount(); i++) {
-                        onViewCreated(parent.getChildAt(i), context, null);
+                    int childCount = parent.getChildCount();
+                    if (childCount != 0) {
+                        // Process children, defer draw as it has set the typeface.
+                        for (int i = 0; i < childCount; i++) {
+                            onViewCreated(parent.getChildAt(i), context, null);
+                        }
                     }
+
+                    // Our dark deed is done
+                    parent.getViewTreeObserver().removeGlobalOnLayoutListener(this);
```

Comment:
Ahh yeah you are right. Didn't think about it like that.

I forgot the reason I worked around it in the first place. :)

OK I'm happy with this then.

On Tue, 27 Jan 2015 14:32 Daniel Lew <notifications@github.com> wrote:

> In
> calligraphy/src/main/java/uk/co/chrisjenx/calligraphy/CalligraphyFactory.java
> <https://github.com/chrisjenx/Calligraphy/pull/121#discussion_r23610892>:
>
> >                      }
> > +
> > +                    // Our dark deed is done
> > +                    parent.getViewTreeObserver().removeGlobalOnLayoutListener(this);
>
> If you inflate the Views, then add them to the Toolbar, you get the
> Calligraphy processing, right?
>
> That feels consistent with the rest of Calligraphy - inflating Views gives
> you Calligraphy, manually creating them doesn't. Adding/removing the Views
> isn't part of the process (though maybe one day it could be - but then it'd
> be best to apply that in a general way to *all* Views, not just the
> Toolbar).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chrisjenx/Calligraphy/pull/121/files#r23610892>.
>